### DISCUSSION TITLE
It would be helpful if we could designate a RoI for tracking

### DISCUSSION THREAD

**Discussion by olinesn** [talmolab/sleap#2015] [Ideas]
Hello,

I'm writing to suggest a feature where the user could fence in the region (in pixels) that sleap-track would run on. This would be analogous to cropping the video to a smaller video.

The use case for me is that sometimes, reflections get picked up as skeletons during inference (see below).

Thanks!
Stefan

![vlcsnap-2024-11-04-10h58m54s842](https://github.com/user-attachments/assets/c34bef85-3b8d-459d-aabf-92cc3c5cad1e)

**Comment by talmo** [comment]
Agreed, this would be a great feature!

Sub-tasks:
1. What kind of ROIs should we allow?
    - Arbitrary polygons?
    - One or many?
2. UX for defining ROIs
3. Data structures to store them
4. What part of the inference pipeline do they get applied to?
    - How do we handle instances that are partially overlapping?
    - After pose or after tracking? Both?
5. How do we handle new videos?
    - Apply ROI manually to all new videos?
    - Automated ROI detection? (New model type?)

Related: #1488 #1388, #1391

**Reply by olinesn** [reply]
re: 1, I like the way that idtracker does it, you just click out some points and accept a drawn ROI, and you can always use text to define or import the same set of XY points

<img width="1512" alt="Screenshot 2024-11-04 at 11 27 26 AM" src="https://github.com/user-attachments/assets/1914668a-dc29-4b76-a5e8-7cb21ae7ae9a">

<img width="1512" alt="Screenshot 2024-11-04 at 11 28 32 AM" src="https://github.com/user-attachments/assets/02290880-9f6a-44b0-8210-549f459bb2e4">
